File: leetcode/golang/1493_longest_subarray_of_1s_after_deleting_one_element.go
func longestSubarray(nums []int) int {
    res := 0
    for l, r, tmp := 0, 0, 0; r < len(nums); r++ {
        tmp += nums[r]
        if tmp < r - l {
            tmp -= nums[l]
            l++
        }
        res = max(res, r - l)
    }
    return res
}

File: leetcode/c++/1493_longest_subarray_of_1s_after_deleting_one_element.cpp
// https://leetcode.cn/problems/longest-subarray-of-1s-after-deleting-one-element/submissions/

#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int res = 0;
        for (int l = 0, r = 0, tmp = 0; r < nums.size(); r++) {
            tmp += nums[r];
            if (tmp < r - l)
                tmp -= nums[l++];
            res = max(res, r - l);
        }
        return res;
    }
};